projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.ts

```diff
@@ -17,31 +18,37 @@ export class MeasurementsPickerComponent implements OnInit {
   allSeriesChecked: boolean = true;
   activeKeywords = 0;
 
+  subscriptions: Subscription = new Subscription();
+
   constructor(private executionPageService: TimeSeriesContextsFactory) {}
 
   ngOnInit(): void {
     if (!this.contextId) {
       throw new Error('Context Id input must be present');
     }
     this.keywordsService = this.executionPageService.getContext(this.contextId).getKeywordsContext();
-    this.keywordsService.onKeywordsUpdated().subscribe((keywords) => {
-      this.keywords = keywords;
-      this.activeKeywords = 0;
-      Object.keys(keywords).forEach((key) => {
-        if (keywords[key].isVisible && keywords[key].isSelected) {
+    this.subscriptions.add(
```

Comment:
Just FYI, another approach to control the subscriptions lifecycle - using the `takeUntil` operator:
```
// Define a subject in the component
private terminator$ = new Subject<unknown>();

...
// Before subscription add takeUntil
this.keywordsService
   .onKeywordsUpdated()
   .pipe(
     takeUntil(this.terminator$)
   )
  .subscribe((keywords) => {
     ...
  });

...
// In destroy hook invoke and complete the subject to terminate the susbscriptions

ngOnDestroy(): void {
    this.terminator$.next();
    this.terminator$.complete();
}
```

From my point of view such approach looks cleaner and easier to use.

Reply:
Thank you. I also like it better. I will use this from now